submit/controllers/ui/new/upload.py

```diff
@@ -577,26 +578,26 @@ def _new_file(params: MultiDict, pointer: FileStorage, session: Session,
     if not form.validate():
         logger.error('Invalid upload form: %s', form.errors)
 
-        alerts.flash_failure("Something went wrong. Please try again.",
-                             title="Whoops")
-        # redirect = url_for('ui.file_upload',
-        #                    submission_id=submission.submission_id)
-        # return {}, status.SEE_OTHER, {'Location': redirect}
-        logger.debug('Invalid form data')
-        raise BadRequest(rdata)
+        alerts.flash_failure(
+            "No file was uploaded; please try again.",
+            title="Something went wrong")
+
+        loc = url_for('ui.file_upload', submission_id=submission.submission_id)
+        return {}, status.SEE_OTHER, {'Location': loc}
```

Comment:
Was the BadRequest causing problems?

The result from the controller should be handled by flow_control.py so I should be able to do a BadReqeust and still get back to the correct page with the upload form.

Reply:
Without additional logic, `BAD_REQUEST` will prevent display of any existing uploaded files.

Reply:
Got it. :+1: